coriolis/tests/osmorphing/test_redhat.py

```diff
@@ -0,0 +1,522 @@
+        mock_test_path_chroot.return_value = True
+
+        result = self.morphing_tools._get_grub2_cfg_location()
+
+        self.assertEqual(result, '/boot/efi/EFI/redhat/grub.cfg')
+        mock_exec_cmd_chroot.assert_has_calls([
+            mock.call("mount /boot || true"),
+            mock.call("mount /boot/efi || true")
+        ])
+        mock_test_path_chroot.assert_called_once_with(
+            '/boot/efi/EFI/redhat/grub.cfg')
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_test_path_chroot')
+    def test__get_grub2_cfg_location_bios(self, mock_test_path_chroot,
+                                          mock_exec_cmd_chroot):
+        mock_test_path_chroot.side_effect = [False, True]
+
+        result = self.morphing_tools._get_grub2_cfg_location()
+
+        mock_test_path_chroot.assert_called_with('/boot/grub2/grub.cfg')
+        mock_exec_cmd_chroot.assert_has_calls([
+            mock.call("mount /boot || true"),
+            mock.call("mount /boot/efi || true")
+        ])
+
+        self.assertEqual(result, '/boot/grub2/grub.cfg')
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_test_path_chroot')
+    def test__get_grub2_cfg_location_unknown(self, mock_test_path_chroot,
+                                             mock_exec_cmd_chroot):
+        mock_test_path_chroot.return_value = False
+
+        self.assertRaisesRegex(
+            Exception,
+            "could not determine grub location. boot partition not mounted?",
+            self.morphing_tools._get_grub2_cfg_location
+        )
+
+        mock_exec_cmd_chroot.assert_has_calls([
+            mock.call("mount /boot || true"),
+            mock.call("mount /boot/efi || true")
+        ])
+        mock_test_path_chroot.assert_has_calls([
+            mock.call('/boot/efi/EFI/redhat/grub.cfg'),
+            mock.call('/boot/grub2/grub.cfg')
+        ])
+
+    @ddt.data(
+        ([("/path/to/ifcfg-file", {"HWADDR": "mac", "NAME": "test-name"})],
+         [], [("test-name", "mac")]),
+        ([("/path/to/ifcfg-file", {"NAME": "test-name"})],
+         ["mac"], [("test-name", "mac")]),
+        ([("/path/to/ifcfg-file", {"HWADDR": "mac"})],
+         ["mac"], [("file", "mac")]),
+        ([("/path/to/ifcfg-file", {"NAME": "test-name"})], [], [])
+    )
+    @ddt.unpack
+    def test__get_net_ifaces_info(self, ifcfgs_ethernet, mac_addresses,
+                                  expected):
+        result = self.morphing_tools._get_net_ifaces_info(
+            ifcfgs_ethernet, mac_addresses)
+        self.assertEqual(result, expected)
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_test_path')
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_write_file_sudo')
+    @mock.patch.object(redhat.utils, 'get_udev_net_rules')
```

Comment:
You don't need to mock `get_udev_net_rules` since it returns a string based on the `net_ifaces_info`:
https://github.com/cloudbase/coriolis/blob/88d55a7e8a843a18a0d03ede1637ce12893bec6e/coriolis/utils.py#L190-L196

Use a proper dict value for `net_ifaces_info`, and adjust the assertions.